compose/material/material/src/desktopMain/kotlin/androidx/compose/material/DesktopMenu.desktop.kt

```diff
@@ -114,26 +116,95 @@ fun DropdownMenu(
             transformOriginState.value = calculateTransformOrigin(parentBounds, menuBounds)
         }
 
-        var focusManager: FocusManager? by mutableStateOf(null)
-        var inputModeManager: InputModeManager? by mutableStateOf(null)
-        Popup(
+        OpenDropdownMenu(
+            expandedStates = expandedStates,
+            popupPositionProvider = popupPositionProvider,
+            transformOriginState = transformOriginState,
+            onDismissRequest = onDismissRequest,
             focusable = focusable,
+            modifier = modifier,
+            content = content
+        )
+    }
+}
+
+/**
+ * A [DropdownMenu] behaves similarly to [Popup] and will use the current position of the mouse
+ * cursor to position itself on screen.
+ *
+ * The [content] of a [DropdownMenu] will typically be [DropdownMenuItem]s, as well as custom
+ * content. Using [DropdownMenuItem]s will result in a menu that matches the Material
+ * specification for menus.
+ *
+ * @param state The open/closed state of the menu.
+ * @param onDismissRequest Called when the user requests to dismiss the menu, such as by
+ * tapping outside the menu's bounds
+ *
+ */
+@Composable
+fun DropdownMenu(
```

Comment:
Could you add the same API for material3 module?

Reply:
Add it how? Just copy/paste?

Reply:
I would say we need to do the same changes - create a state, and expose it instead of popupPositionProvider, which we also have there.

The implmentation itself is different there, but API almost the same.

Reply:
We don't currently have any material3 desktop APIs. The only file in `material3/desktopMain` is `NavigationKeyEvents.desktop.kt`.
So to add it would mean to copy/paste all the code from `material/DropdownMenu.desktop.kt` and anything in `material` it uses, transitively.

Reply:
Understood. We should commonise this (not in this PR). At least in skikoMain. We also have menus in uikitMain